src/transformers/models/encoder_decoder/modeling_flax_encoder_decoder.py

```diff
@@ -0,0 +1,819 @@
+                decoder_input_ids,
+                decoder_attention_mask,
+                decoder_position_ids,
+                **kwargs,
+            )
+
+        outputs = self.module.apply(
+            inputs,
+            decoder_input_ids=jnp.array(decoder_input_ids, dtype="i4"),
+            decoder_attention_mask=jnp.array(decoder_attention_mask, dtype="i4"),
+            decoder_position_ids=jnp.array(decoder_position_ids, dtype="i4"),
+            encoder_hidden_states=encoder_hidden_states,
+            encoder_attention_mask=jnp.array(encoder_attention_mask, dtype="i4"),
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_dict=return_dict,
+            deterministic=not train,
+            rngs=rngs,
+            mutable=mutable,
+            method=_decoder_forward,
+        )
+
+        # add updated cache to model output
+        if past_key_values is not None and return_dict:
+            outputs, past = outputs
+            outputs["past_key_values"] = unfreeze(past["cache"])
+            return outputs
+        elif past_key_values is not None and not return_dict:
+            outputs, past = outputs
+            outputs = outputs[:1] + (unfreeze(past["cache"]),) + outputs[1:]
+
+        return outputs
+
+    @add_start_docstrings_to_model_forward(ENCODER_DECODER_INPUTS_DOCSTRING)
+    @replace_return_docstrings(output_type=FlaxSeq2SeqLMOutput, config_class=_CONFIG_FOR_DOC)
+    def __call__(
+        self,
+        input_ids: jnp.ndarray,
+        attention_mask: Optional[jnp.ndarray] = None,
+        decoder_input_ids: Optional[jnp.ndarray] = None,
+        decoder_attention_mask: Optional[jnp.ndarray] = None,
+        position_ids: Optional[jnp.ndarray] = None,
+        decoder_position_ids: Optional[jnp.ndarray] = None,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+        train: bool = False,
+        params: dict = None,
+        dropout_rng: PRNGKey = None,
+    ):
+        r"""
+        Returns:
+
+        Examples::
+
+            >>> from transformers import FlaxEncoderDecoderModel, GPT2Tokenizer
+
+            >>> # initialize a gpt2togpt2 from pretrained GPT2 models. Note that the cross-attention layers will be randomly initialized
+            >>> model = FlaxEncoderDecoderModel.from_encoder_decoder_pretrained('gpt2', 'gpt2')
+
+            >>> tokenizer = GPT2Tokenizer.from_pretrained('gpt2')
+
+            >>> # forward
+            >>> input_ids = tokenizer.encode("Hello, my dog is cute", add_special_tokens=True, return_tensors='jax')  # Batch size 1
+            >>> outputs = model(input_ids=input_ids, decoder_input_ids=input_ids)
+
+            >>> # save and load from pretrained
+            >>> model.save_pretrained("gpt2togpt2")
+            >>> model = FlaxEncoderDecoderModel.from_pretrained("gpt2togpt2")
+
```

Comment:
we could actually show a complete generation example here, *e.g.* the one I added to the tests

Reply:
This example won't give any useful result at the moment as the cross attention weights are randomly initialized. Could we maybe instead use `"patrickvonplaten/bert2gpt2-cnn_dailymail-fp16"` with a sensible text (maybe a 2 sentence text) and put the generated answer in a comment? :-)

I just uploaded the flax weights so `.from_pretrained("patrickvonplaten/bert2gpt2-cnn_dailymail-fp16")` should now work

Reply:
https://huggingface.co/patrickvonplaten/bert2gpt2-cnn_dailymail-fp16/commit/a30547834c8e029c92e00ece8402b210ec4aa2a9